File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left=0
        right=len(height)-1
        ma=0
        
        while(left<right):
            lbar=height[left]
            rbar=height[right]
            
            ma=max(ma,(min(lbar,rbar)*(right-left)))
            
            if(lbar<rbar):
                left+=1
            else:
                right-=1
        
        return ma
        
        
            

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) 
    {
        int t=h.size();
        int i=0,j=t-1;
        int area=0;
        
        
    while(i<j)
    {
        int m=min(h[i],h[j]);
        area=max(area,m*(j-i));
        if(h[i]<h[j])
        {
            i++;
            
        }
        else
            j--;
        
    }return area;
}
};